### PR TITLE
DBZ-4463 Upgrade to SQL Server driver 9.x

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-4463

### CODE REVIEW

**Comment by jpechane** [comment]
Vitess failure fixed via https://github.com/debezium/debezium-connector-vitess/pull/62

**Comment by jpechane** [comment]
POstgres/server/REST extension failures solved via https://github.com/debezium/debezium/pull/3060

**Comment by jpechane** [comment]
Db2 failure solved via https://github.com/debezium/debezium/pull/3061

File: pom.xml [hunk 1]
```diff
@@ -120,7 +120,7 @@
         <version.mysql.driver>8.0.27</version.mysql.driver>
         <version.mysql.binlog>0.25.4</version.mysql.binlog>
         <version.mongo.driver>4.3.3</version.mongo.driver>
-        <version.sqlserver.driver>7.2.2.jre8</version.sqlserver.driver>
+        <version.sqlserver.driver>9.4.1.jre11</version.sqlserver.driver>
```
Comment (gunnarmorling) [inline pom.xml:123]: jre11 seems to indicate that would not run on Java 1.8? Didn't we mean to keep that for DBZ 2.0?

Comment (jpechane) [reply pom.xml]: @gunnarmorling Changed, I forgot that we are on Java 11 for building but Java 8 for runtime.

**Comment by gunnarmorling** [comment]
@jpechane, pushed a commit for handling the Vitess issue.

**Comment by gunnarmorling** [comment]
Let's make it a habit to check the license on any such major version bump of a dependency and put a comment on it to the PR or issue (I wouldn't expect changes in minor/micro updates). I just did it, it's still using MIT, i.e. all good. Thx!

**Pull request merged**